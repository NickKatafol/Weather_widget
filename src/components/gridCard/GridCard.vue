<script setup lang="ts">
import {computed, reactive, toRef, toRefs} from "vue"
import type {ICardCell} from "./gridCardTypes"

const cardProps = defineProps<{
  cellSets: ICardCell[]
}>()

function cellStyle (cellSet: ICardCell) {
  return {
    gridRow: cellSet.row,
    gridColumn: cellSet.column,
    backgroundImage: cellSet.imgName ? `url(${cellSet.pathToImgResource}${cellSet.imgName})` : '',
    border: cellSet.imgName ? 'none' : ''
  }
}

</script>


<template>
  <div class="weather-card-wrapper">
    <div v-for="(cellSet, ind) of cardProps.cellSets"
         :key="'set' + ind"
         :style="[cellStyle(cellSet), cellSet.cellStyles]"
         class="cell_item"
    >
      <div>{{ cellSet.leftVal }}</div>
      <div>{{ cellSet.rightVal }}</div>
    </div>
  </div>
</template>


<style scoped lang="scss">
//grid application
.weather-card-wrapper {
  width: 100%;
  border-left: var(--vt-c-grey_dark) 1px solid;
  border-bottom: var(--vt-c-grey_dark) 1px solid;
  box-sizing: border-box;
  padding: var(--vt-c-common-padding);
  
  display: grid;
  grid-auto-rows: 30px;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: var(--vt-c-common-padding);
  grid-auto-flow: row dense;
  
  .cell_item {
    box-sizing: border-box;
    padding: var(--vt-c-common-padding) calc(var(--vt-c-common-padding) / 2);
    border-left: var(--vt-c-grey_dark) 1px solid;
    
    display: flex;
    align-items: center;
    justify-content: space-between;
    
    background: no-repeat;
    background-size: contain;
    background-position: right top;
  }
}
</style>