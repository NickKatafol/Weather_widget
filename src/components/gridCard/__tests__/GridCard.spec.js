import { describe, it, expect } from 'vitest'
import {render} from "@testing-library/vue";

import GridCard from '../GridCard.vue'
import {fakeGridCardCellSets} from "./fakeGridCardProps";

describe('Render component with props', () => {
  it('GridCard is rendering properly', () => {
    const wrapper = render(GridCard, {
      props: {cellSets: fakeGridCardCellSets}
    })

    expect(wrapper).toMatchSnapshot()
  })
})
