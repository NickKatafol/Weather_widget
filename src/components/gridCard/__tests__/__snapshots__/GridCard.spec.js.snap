// Vitest Snapshot v1

exports[`Render component with props > GridCard is rendering properly 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="weather-card-wrapper"
        data-v-5f9c2d4f=""
      >
        
        <div
          class="cell_item"
          data-v-5f9c2d4f=""
          style="grid-row: 1; grid-column: 1; color: rgb(177, 114, 193); border-color: #b172c1;"
        >
          <div
            data-v-5f9c2d4f=""
          >
            Miami
          </div>
          <div
            data-v-5f9c2d4f=""
          >
            US
          </div>
        </div>
        <div
          class="cell_item"
          data-v-5f9c2d4f=""
          style="grid-row: 2; grid-column: 1;"
        >
          <div
            data-v-5f9c2d4f=""
          >
            longitude
          </div>
          <div
            data-v-5f9c2d4f=""
          >
            -80.1936
          </div>
        </div>
        <div
          class="cell_item"
          data-v-5f9c2d4f=""
          style="grid-row: 3; grid-column: 1;"
        >
          <div
            data-v-5f9c2d4f=""
          >
            latitude
          </div>
          <div
            data-v-5f9c2d4f=""
          >
            25.7742
          </div>
        </div>
        <div
          class="cell_item"
          data-v-5f9c2d4f=""
          style="grid-row: 5; grid-column: 1;"
        >
          <div
            data-v-5f9c2d4f=""
          >
            temperature
          </div>
          <div
            data-v-5f9c2d4f=""
          >
            86°F
          </div>
        </div>
        <div
          class="cell_item"
          data-v-5f9c2d4f=""
          style="grid-row: 5; grid-column: 2;"
        >
          <div
            data-v-5f9c2d4f=""
          >
            pressure
          </div>
          <div
            data-v-5f9c2d4f=""
          >
            1018 hPa
          </div>
        </div>
        <div
          class="cell_item"
          data-v-5f9c2d4f=""
          style="grid-row: 1/span 3; grid-column: 2; background-image: url(/src/domains/Weather/assets/imgs/weatherImgs/sun.png);"
        >
          <div
            data-v-5f9c2d4f=""
          />
          <div
            data-v-5f9c2d4f=""
          />
        </div>
        
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="weather-card-wrapper"
      data-v-5f9c2d4f=""
    >
      
      <div
        class="cell_item"
        data-v-5f9c2d4f=""
        style="grid-row: 1; grid-column: 1; color: rgb(177, 114, 193); border-color: #b172c1;"
      >
        <div
          data-v-5f9c2d4f=""
        >
          Miami
        </div>
        <div
          data-v-5f9c2d4f=""
        >
          US
        </div>
      </div>
      <div
        class="cell_item"
        data-v-5f9c2d4f=""
        style="grid-row: 2; grid-column: 1;"
      >
        <div
          data-v-5f9c2d4f=""
        >
          longitude
        </div>
        <div
          data-v-5f9c2d4f=""
        >
          -80.1936
        </div>
      </div>
      <div
        class="cell_item"
        data-v-5f9c2d4f=""
        style="grid-row: 3; grid-column: 1;"
      >
        <div
          data-v-5f9c2d4f=""
        >
          latitude
        </div>
        <div
          data-v-5f9c2d4f=""
        >
          25.7742
        </div>
      </div>
      <div
        class="cell_item"
        data-v-5f9c2d4f=""
        style="grid-row: 5; grid-column: 1;"
      >
        <div
          data-v-5f9c2d4f=""
        >
          temperature
        </div>
        <div
          data-v-5f9c2d4f=""
        >
          86°F
        </div>
      </div>
      <div
        class="cell_item"
        data-v-5f9c2d4f=""
        style="grid-row: 5; grid-column: 2;"
      >
        <div
          data-v-5f9c2d4f=""
        >
          pressure
        </div>
        <div
          data-v-5f9c2d4f=""
        >
          1018 hPa
        </div>
      </div>
      <div
        class="cell_item"
        data-v-5f9c2d4f=""
        style="grid-row: 1/span 3; grid-column: 2; background-image: url(/src/domains/Weather/assets/imgs/weatherImgs/sun.png);"
      >
        <div
          data-v-5f9c2d4f=""
        />
        <div
          data-v-5f9c2d4f=""
        />
      </div>
      
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
